src/System.Runtime.Numerics/tests/BigInteger/multiply.cs

```diff
@@ -38,8 +38,16 @@ public static void RunMultiply_TwoLargeBigIntegers()
         public static void RunMultiply_TwoLargeBigIntegers_Threshold()
         {
             // Again, with lower threshold
-            BigIntTools.Utils.RunWithFakeThreshold("SquareThreshold", 8, RunMultiply_TwoLargeBigIntegers);
-            BigIntTools.Utils.RunWithFakeThreshold("MultiplyThreshold", 8, RunMultiply_TwoLargeBigIntegers);
+            BigIntTools.Utils.RunWithFakeThreshold("SquareThreshold", 8, () =>
+                BigIntTools.Utils.RunWithFakeThreshold("MultiplyThreshold", 8, RunMultiply_TwoLargeBigIntegers)
+            );
+
+            // Again, with lower threshold
+            BigIntTools.Utils.RunWithFakeThreshold("SquareThreshold", 8, () =>
+                BigIntTools.Utils.RunWithFakeThreshold("MultiplyThreshold", 8, () =>
+                    BigIntTools.Utils.RunWithFakeThreshold("AllocationThreshold", 8, RunMultiply_TwoLargeBigIntegers)
```

Comment:
There are two runs to cover all the code: the first run modifies only the square and multiply thresholds to execute the stackalloc based code, the second run modifies all three to execute the array based code.

But it should be valid too, if we modify the allocation threshold to say 16 and do only one run.